ui/components/multichain/asset-picker-amount/asset-picker-amount.js

```diff
@@ -0,0 +1,104 @@
+import AssetPicker from './asset-picker/asset-picker';
+import MaxClearButton from './max-clear-button/max-clear-button';
+
+// A component that combines an asset picker with an input for the amount to send.
+// Work in progress.
+export default function AssetPickerAmount() {
+  const dispatch = useDispatch();
+  const t = useI18nContext();
+  const asset = useSelector(getSendAsset);
+  const amount = useSelector(getSendAmount);
+  const inError = useSelector(sendAmountIsInError);
+  const transaction = useSelector(getCurrentDraftTransaction);
+  const nativeCurrency = useSelector(getNativeCurrency);
+  const nativeCurrencyImage = useSelector(getNativeCurrencyImage);
+  const tokenList = useSelector(getTokenList);
+
+  let balance = new Numeric(transaction.asset.balance, 16).toBase(10);
+  if (asset.type === AssetType.native) {
+    balance = balance.applyConversionRate(1 / Math.pow(10, 18));
+  }
+
+  // TODO: Handle really long symbols in the UI
+  const symbol =
+    asset.type === AssetType.native ? nativeCurrency : asset.details?.symbol;
+
+  const image =
+    asset.type === AssetType.native
+      ? nativeCurrencyImage
+      : asset.details.image ??
+        tokenList?.[asset.details.address.toLowerCase()]?.iconUrl;
+
+  return (
+    // TODO: Error state when insufficient funds
+    <Box className="asset-picker-amount">
+      <Label className="asset-picker-amount__amount">{t('amount')}</Label>
+      <MaxClearButton />
+      <Box className="asset-picker-amount__box" borderRadius={BorderRadius.LG}>
+        <AssetPicker asset={{ symbol, image }} />
+
+        {/* TODO: See if the native and token inputs can be merged into 1 component */}
+        {asset.type === AssetType.native ? (
+          <UserPreferencedCurrencyInput
+            onChange={(newAmount) => dispatch(updateSendAmount(newAmount))}
+            hexValue={amount}
+            error={inError}
+            className="asset-picker-amount__input"
+            swapIcon={(onClick) => (
+              <Icon
+                className="asset-picker-amount__swap-icon"
+                name={IconName.SwapVertical}
+                size={IconSize.Sm}
+                color={IconColor.primaryDefault}
+                onClick={onClick}
+              />
+            )}
+          />
+        ) : (
+          <UserPreferencedTokenInput
+            error={inError}
+            onChange={(newAmount) => dispatch(updateSendAmount(newAmount))}
+            token={asset.details}
+            value={amount}
+            className="asset-picker-amount__input"
+            // TODO: Consider supporting swapping currencies for tokens so you can
+            // type fiat values. (But only when a conversion rate is available)
+          />
+        )}
+      </Box>
+      <Text className="asset-picker-amount__balance">
+        {/* TODO: Consider rounding the balance so its not super long? */}
+        {t('balance')}: {balance.toString()} {symbol}
+      </Text>
+    </Box>
+  );
+}
```

Comment:
Could we get a storybook file for these new components? Docs here https://metamask.github.io/metamask-storybook/?path=/docs/getting-started-documentation-guidelines--docs#creating-a-story or checkout any examples in the multichain folder or component-library folder

Reply:
Added some stories: https://github.com/MetaMask/metamask-extension/pull/21517/commits/470e5da8ff2fa53747b38707b27014708c3a766b#diff-5960d110edf2961573243a029e09fb3475cc831ff908448c9c37784e3d569301